sotrplib/maps/core.py

```diff
@@ -596,7 +631,7 @@ def build(self):
             self.get_time_and_mapdepth(sourcemap)
             self.update_map_times(sourcemap)
 
-        if not isinstance(self.time, type(None)):
+        if not isinstance(self.time_mean, type(None)):
```

Comment:
if self.time_mean is not None

Reply:
Yep works for any object. None is a singleton and all you're doing is asking if the 'time_mean' object is exactly the None object. Which it won't be if it's an array.